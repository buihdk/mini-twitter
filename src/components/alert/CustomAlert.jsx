import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

import Alert from 'react-bootstrap/Alert';

import './CustomAlert.scss';

const CustomAlert = ({ isShown, variant, message }) => (
  <Alert
    className={clsx('customAlert', isShown ? 'shown' : 'hidden')}
    variant={variant}
  >
    {message}
  </Alert>
);

CustomAlert.propTypes = {
  isShown: PropTypes.string,
  variant: PropTypes.string,
  message: PropTypes.string,
};

CustomAlert.defaultProps = {
  isShown: '',
  variant: '',
  message: '',
};

export default CustomAlert;
