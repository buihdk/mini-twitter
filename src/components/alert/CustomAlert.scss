.custom-alert {
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  border-radius: unset;
  z-index: 99;
}

.shown {
  opacity: 0.85;
  transition: all 250ms linear;
}

.hidden {
  opacity: 0;
  transition: all 250ms linear;
}
