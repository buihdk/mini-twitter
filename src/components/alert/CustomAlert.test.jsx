import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import CustomAlert from './CustomAlert';

describe('CustomAlert', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(
      <CustomAlert isShown="isTweet" variant="success" message="test" />,
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
