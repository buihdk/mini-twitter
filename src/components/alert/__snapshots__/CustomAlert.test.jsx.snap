// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAlert renders without crashing 1`] = `
<Alert
  className="custom-alert shown"
  closeLabel="Close alert"
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "displayName": "Fade",
      "render": [Function],
    }
  }
  variant="success"
>
  test
</Alert>
`;
